src/structureUtilities.jl

```diff
@@ -10,15 +10,134 @@ Arguments:
 * `algo`: Algorithm, one out of [:learnspn, :random].
 """
 function generate_spn(X::Matrix, algo::Symbol; params...)
-
     if algo == :learnspn
         return learnspn(X; params...)
     elseif algo == :random
         return randomspn(X; params...)
     else
-        @error("Unknown structure learning method: ", method)
+        @error("Unknown structure learning algorithm: ", algo)
+    end
+end
+
+"""
+    learnspn(X; distribution=Normal(), minclustersize=100)
+
+Return Sum Product Network learned by a simplified LearnSPN algorithm.
+"""
+function learnspn(X; distribution=Normal(), minclustersize=100)
```

Comment:
I suggest that `distribution` is an argument and not a keyword argument and is instead a vector of distributions. Something like this:

```Julia
learnspn(X, distributions::AbstractVector{Type{D}}; ...) where {D<:UnivariateDistribution}
```